src/utils.js

```diff
@@ -1,56 +1,36 @@
 "use strict";
 
-const axios = require("axios");
 const path = require("path");
 const fs = require("fs");
 const urljoin = require("url-join");
 
-const defaultPortalUrl = "https://siasky.net";
+/**
+ * The default URL of the Skynet portal to use in the absence of configuration.
+ */
+const defaultSkynetPortalUrl = "https://siasky.net";
 
+/**
+ * The URI prefix for Skynet.
+ */
 const uriSkynetPrefix = "sia://";
 
 function defaultOptions(endpointPath) {
   return {
-    portalUrl: defaultPortalUrl,
+    portalUrl: defaultPortalUrl(),
     endpointPath: endpointPath,
 
     APIKey: "",
     customUserAgent: "",
   };
 }
 
+// TODO: This will be smarter. See
+// https://github.com/NebulousLabs/skynet-docs/issues/21.
 /**
- * Creates and executes a request.
- * @param {Object} config - Configuration for the request.
- * @param {string} config.method - HTTP method to use.
- * @param {string} [config.APIKey] - Authentication password to use.
- * @param {string} [config.customUserAgent=""] - Custom user agent header to set.
- * @param {Object} [config.data=null] - Data to send in a POST.
- * @param {string} [config.endpointPath=""] - The relative URL path of the portal endpoint to contact.
- * @param {string} [config.extraPath=""] - Extra path element to append to the URL.
- * @param {Function} [config.onUploadProgress] - Optional callback to track progress.
- * @param {Object} [config.params={}] - Query parameters to include in the URl.
- * @param {string} [config.responseType=""] - The type of data that the server will respond with.
+ * Selects the default portal URL to use when initializing a client. May involve network queries to several candidate portals.
  */
-function executeRequest(config) {
-  const url = makeUrl(config.portalUrl, config.endpointPath, config.extraPath ? config.extraPath : "");
-
-  return axios({
-    url,
-    method: config.method,
-    data: config.data,
-    params: config.params,
-    headers: config.customUserAgent && { "User-Agent": config.customUserAgent },
-    auth: config.APIKey && { username: "", password: config.APIKey },
-    responseType: config.responseType,
-    onUploadProgress:
-      config.onUploadProgress &&
-      function ({ loaded, total }) {
-        const progress = loaded / total;
-
-        config.onUploadProgress(progress, { loaded, total });
-      },
-  });
+function defaultPortalUrl() {
```

Comment:
why is this a function now?

Reply:
See https://github.com/NebulousLabs/nodejs-skynet/pull/56#discussion_r476475082 and https://github.com/NebulousLabs/skynet-docs/issues/21